### PR TITLE
[WIP] Webpack

### PR DESCRIPTION
**Pull request by DaanDD**
I had a first go at implementing Webpack in our frontend build process. Compiling sass code through Webpack was extremely slow, because it can only build to a js file, which then needs to be extracted to a CSS file with extract-text-webpack-plugin. So I opted to implement a gulp build process, with webpack as a separate task. Everything should work the same as before, albeit a lot faster. Bootstrap now builds in ~600ms instead of 3s with Grunt.

In the index.js I currently implemented all functionality that was in theme.coffee. The local scroll was easily implemented with the sweet-scroll plugin with only two lines of code, and adding the `img-responsive` class to all images has also been added, although I would suggest doing that through css.

I would also suggest to keep the initial js as small as possible, so I would leave out the sweet-scroll plugin. If it is needed in a project it can be added with 2 lines of code. I also left out the requestAnimationFrame fallback because it can easily be found on npm.

So now everything can be easily built to 1 bundle.js. Fork still loads utils.js and frontend.js which are probably redundant, but I think this is probably part of the Fork core.

### CODE REVIEW

**Comment by WouterSioen** [comment]
looking good :+1:

**Comment by WouterSioen** [comment]
moeten deploy scripts niet aangepast worden om nu `gulp build` te doen ipv `grunt build`?

File: src/Frontend/Themes/Bootstrap/src/Js/index.js [hunk 1]
```diff
@@ -0,0 +1 @@
+console.log('Hello, world!')
```
Comment (carakas) [inline src/Frontend/Themes/Bootstrap/src/Js/index.js]: did you forget to remove a console.log

Comment (DaanDD) [inline src/Frontend/Themes/Bootstrap/src/Js/index.js]: This is outdated, it is already removed.

File: package.json [hunk 2]
```diff
@@ -2,25 +2,43 @@
...
-    "grunt-contrib-uglify": "^0.11.0",
-    "grunt-contrib-watch": "^0.6.1",
-    "grunt-fontgen": "^0.2.3",
-    "grunt-replace": "^0.11.0",
-    "grunt-svgmin": "^3.1.0",
-    "grunt-sync": "^0.5.1",
-    "grunt-usemin": "^3.1.1",
-    "grunt-webfont": "^1.1.0",
-    "load-grunt-tasks": "^3.4.0"
+    "babel-core": "^6.6.4",
```
Comment (carakas) [inline package.json]: is there a reason you didn't just use babel-cli ?

Comment (DaanDD) [inline package.json]: babel-cli isn't necessary here, as it is used through Webpack. Webpack needs babel-core and babel-loader instead.

**Comment by WouterSioen** [comment]
:shipit:

**Comment by tijsverkoyen** [comment]
👍

**Pull request merged**